lib/datasets.go

```diff
@@ -1100,31 +1094,36 @@ func (m *DatasetMethods) Remove(ctx context.Context, p *RemoveParams) (*RemoveRe
 		return nil, fmt.Errorf("can only remove whole dataset versions, not individual components")
 	}
 
-	ref, err := repo.ParseDatasetRef(p.Ref)
+	ref, _, err := m.inst.ParseAndResolveRefWithWorkingDir(ctx, p.Ref, "local")
 	if err != nil {
-		return nil, err
-	}
-
-	if canonErr := repo.CanonicalizeDatasetRef(ctx, m.inst.repo, &ref); canonErr != nil && canonErr != repo.ErrNoHistory {
-		log.Debugf("Remove, repo.CanonicalizeDatasetRef failed, error: %s", canonErr)
-		if p.Force {
-			didRemove, _ := base.RemoveEntireDataset(ctx, m.inst.repo, reporef.ConvertToDsref(ref), []dsref.VersionInfo{})
+		log.Debugw("Remove, repo.ParseAndResolveRefWithWorkingDir failed", "err", err)
+		// TODO(b5): this "logbook.ErrNotFound" is needed to get cmd.TestRemoveEvenIfLogbookGone
```

Comment:
This behavior is pretty old, it was implemented in https://github.com/qri-io/qri/commit/fe4598abaa9f4427f05ae8a046e9dcc1fc9a318e. Back then, adding a dataset using `qri add` would not write anything to the logbook, which made it impossible to remove datasets from the refstore. That's not the case anymore, so I think it's no longer necessary to bend over backwards to support this out-of-date functionality.